src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -870,7 +872,13 @@ static bool GetPlatformNames(SmallDictionary<string, Versions> attributes, Small
                 return platformNames;
             }
 
-            static string GetFormattedString(string resource, params object[] args) => string.Format(CultureInfo.InvariantCulture, resource, args);
+            static string GetFormattedString(string resource, string platformName, object? arg1 = null, object? arg2 = null) =>
+                string.Format(CultureInfo.InvariantCulture, resource, AddOsxIfNeeded(platformName), arg1, arg2);
+
+            static string AddOsxIfNeeded(string platformName) =>
+                platformName.Equals(macOS, StringComparison.OrdinalIgnoreCase) ? MacSlashOSX : platformName;
+
+            static string EncloseWithQuota(string pName) => $"'{AddOsxIfNeeded(pName)}'";
```

Comment:
This was probably meant to be named `EncloseWithQuotes`. This should also use a resource string since `"` characters are not universal and should be localizable. French uses `«{0}»`, for example.

Reply:
> This was probably meant to be named EncloseWithQuotes. 

Right

> This should also use a resource string since " characters are not universal and should be localizable. French uses «{0}», for example.

I am not sure that is necessary, in this repo we every use single quotes for enclosing APIs  `'{0}'` everywhere. https://github.com/dotnet/roslyn-analyzers/blob/fde0c6667d5af4b8cf1c7b135325748f2c0fb226/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx#L124

Reply:
i think `'{0}'` is convention even in runtime

Reply:
OK fair enough. It looks like our [French translation of that message](https://github.com/dotnet/roslyn-analyzers/blob/fde0c6667d5af4b8cf1c7b135325748f2c0fb226/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/xlf/MicrosoftNetCoreAnalyzersResources.fr.xlf#L1870-L1874) does not change the quotation marks. Thanks for checking it.